api/generate-api-docs.js

```diff
@@ -1,60 +1,161 @@
 
-// Execute a command as if we were in the terminal
-function runCommand(command) {
-  const result = execSync(command, { stdio: "inherit" });
-  return result;
-}
-
 // API name must be passed in as an arg.
-const api = process.argv[2];
-if (api === undefined) {
+const requestedAPI = process.argv[2];
+if (requestedAPI === undefined) {
   const validAPIs = Object.keys(apiStrategies).join(", ");
   console.log(`Please specify an API name. Valid names: ${validAPIs}`);
   process.exit();
 }
 
 // The API name must be recognized.
-const strategy = apiStrategies[api];
+const strategy = apiStrategies[requestedAPI];
 if (strategy === undefined) {
   const validAPIs = Object.keys(apiStrategies).join(", ");
-  console.error(`Invalid API name ${api}. Valid names: ${validAPIs}`);
+  console.error(`Invalid API name ${requestedAPI}. Valid names: ${validAPIs}`);
+  process.exit();
+}
+
+// Version is an optional argument. If not provided, we assume "vNext".
+const requestedVersion = process.argv[3];
+
+// Hack: zeebe API is removed at version 8.7, don't allow regeneration in vNext.
+if (requestedAPI === "zeebe" && requestedVersion === undefined) {
+  console.error("Zeebe API docs are no longer in active development.");
   process.exit();
 }
 
+// Find the corresponding configuration in docusaurus.config.js.
+const configs = loadAPIConfigs();
+const apiConfig = buildAPIConfig(configs, requestedAPI, requestedVersion);
+
 // All APIs will execute these same steps, with custom-per-API steps
 //   implemented by each API's generation-strategy.js.
 const steps = [
   // Remove old docs
-  () => runCommand(`docusaurus clean-api-docs ${api} -p api-${api}-openapi`),
+  () => runCommand(buildCleanCommand(apiConfig)),
 
   // Run any custom steps before generation
-  strategy.preGenerateDocs,
+  () => strategy.preGenerateDocs(apiConfig),
 
   // Generate the docs
-  () => runCommand(`docusaurus gen-api-docs ${api} -p api-${api}-openapi`),
+  () => runCommand(buildGenerateCommand(apiConfig)),
 
   // Run any custom steps after generation
-  strategy.postGenerateDocs,
+  () => strategy.postGenerateDocs(apiConfig),
 
   // Run prettier against the generated docs. Twice. Yes, twice.
   //   I don't know why, but the first run always leaves an extra blank line,
   //   which the second execution removes.
-  () => runCommand(`prettier --write ${strategy.outputDir}`),
-  () => runCommand(`prettier --write ${strategy.outputDir}`),
+  () => runCommand(`prettier --write ${apiConfig.outputDir}`),
+  () => runCommand(`prettier --write ${apiConfig.outputDir}`),
 ];
 
 // Run the steps!
 steps.forEach((step) => step());
+
+// ---------- vvvvvvv helper functions vvvvvvv ----------
+
+// Execute a command as if we were in the terminal
+function runCommand(command) {
+  const result = execSync(command, { stdio: "inherit" });
+  return result;
+}
+
+// Load the API configs from the docusaurus.config.js file.
+function loadAPIConfigs() {
+  const config = require("../docusaurus.config");
```

Comment:
Everything we were using in the individual generation strategies is already in the docusaurus config!